sdk/eventgrid/eventgrid-system-events/src/predicates.ts

```diff
@@ -0,0 +1,693 @@
+  "Microsoft.AVS.ClusterDeleted": AvsClusterDeletedEventData;
+  /** An interface for the event data of a "Microsoft.AVS.ClusterUpdating" event. */
+  "Microsoft.AVS.ClusterUpdating": AvsClusterUpdatingEventData;
+  /** An interface for the event data of a "Microsoft.AVS.ClusterUpdated" event. */
+  "Microsoft.AVS.ClusterUpdated": AvsClusterUpdatedEventData;
+  /** An interface for the event data of a "Microsoft.AVS.ClusterFailed" event. */
+  "Microsoft.AVS.ClusterFailed": AvsClusterFailedEventData;
+  /** An interface for the event data of a "Microsoft.ApiCenter.ApiDefinitionAdded" event. */
+  "Microsoft.ApiCenter.ApiDefinitionAdded": ApiCenterApiDefinitionAddedEventData;
+  /** An interface for the event data of a "Microsoft.ApiCenter.ApiDefinitionUpdated" event. */
+  "Microsoft.ApiCenter.ApiDefinitionUpdated": ApiCenterApiDefinitionUpdatedEventData;
+  /** An interface for the event data of a "Microsoft.Communication.AdvancedMessageDeliveryStatusUpdated" event. */
+  "Microsoft.Communication.AdvancedMessageDeliveryStatusUpdated": AcsMessageDeliveryStatusUpdatedEventData;
+  /** An interface for the event data of a "Microsoft.Communication.AdvancedMessageReceived" event. */
+  "Microsoft.Communication.AdvancedMessageReceived": AcsMessageReceivedEventData;
+  /** An interface for the event data of a "Microsoft.Communication.RouterWorkerUpdated" event. */
+  "Microsoft.Communication.RouterWorkerUpdated": AcsRouterWorkerUpdatedEventData;
+}
+
+/**
+ * isCloudEventLike returns "true" when the event is a CloudEvent
+ *
+ * @param o - Either an EventGrid our CloudEvent event.
+ */
+function isCloudEventLike(
+  o: EventGridEvent<unknown> | CloudEvent<unknown>,
+): o is CloudEvent<unknown> {
+  return (o as any).source !== undefined;
+}
+
+/**
+ * iSystemEvent returns "true" when a given event is a system event of a given type. When using
+ * TypeScript, this function acts as a custom type guard and allows the TypeScript compiler to
+ * identify the underlying data
+ *
+ * @param eventType - The type of system event to check for, e.g., "Microsoft.AppConfiguration.KeyValueDeleted"
+ * @param event - The event to test.
+ */
+export function isSystemEvent<T extends KnownSystemEventTypes>(
+  eventType: T,
+  event: EventGridEvent<unknown>,
+): event is EventGridEvent<SystemEventNameToEventData[T]>;
+
+/**
+ * iSystemEvent returns "true" when a given event is a system event of a given type. When using
+ * TypeScript, this function acts as a custom type guard and allows the TypeScript compiler to
+ * identify the underlying data
+ *
+ * @param eventType - The type of system event to check for, e.g., "Microsoft.AppConfiguration.KeyValueDeleted"
+ * @param event - The event to test.
+ */
+export function isSystemEvent<T extends KnownSystemEventTypes>(
+  eventType: T,
+  event: CloudEvent<unknown>,
+): event is CloudEvent<SystemEventNameToEventData[T]>;
+
+export function isSystemEvent<T extends KnownSystemEventTypes>(
```

Comment:
is it expected that `isSystemEvent` is not exposed in public api surface?

Reply:
I have added the missed methods